### DISCUSSION TITLE
Error Simulating Range dispersal: "*** Extents of Species Distribution map do not match headers of LandscapeFile"

### DISCUSSION THREAD

**Discussion by Jeankwizera** [RangeShifter/RangeshiftR-tutorials#17] [Q&A] [answered]
Hello, I am having a challenge simulating range dispersal using my landscape and spDist files. I have a landscape habitat map file saved as ".txt" file, with the header and first few rows in the following format:

NCOLS 1
NROWS 16196
xllcorner 73.25117
yllcorner -54.74922
CELLSIZE 1000
NODATA_VALUE 999
"lat" "lon" "cakile.edentula_AllData_Full.logistic.values" "optional"
-77.75 60.25 5.991846e-05 TRUE
-77.25 60.25 6.092786e-05 TRUE
-77.75 59.75 6.054509e-05 TRUE
-77.25 59.75 6.0859555e-05 TRUE
-76.75 59.75 6.2298204e-05 TRUE
-76.25 59.75 6.296467e-05 TRUE

I also have a species distribution file in a similar format i.e with the header and first few rows in the following format:

NCOLS 5
NROWS 4
xllcorner 73.25117
yllcorner -54.74922
CELLSIZE 1000
NODATA_VALUE 999
"species" "lon" "lat" "uncertainty" "date" "year" "month" "optional"
"species name" 151.2489 -33.8462 NA NA NA NA TRUE
"species name" 144.9125 -37.8332 NA NA NA NA TRUE
"species name" 153.1679 -27.3837 NA NA NA NA TRUE
"species name" 115.8575 -31.9596 NA NA NA NA TRUE

NOTE: I created the landscape and species distribution maps from CSV files and added the headers manually. 

I have set up the landscape, species and initialisation parameters and used them to define the RS object which is validated as TRUE using the function "validateRSparams ( )". The code I have used is as follows:

```{r, echo=TRUE, warning=FALSE, message=FALSE}
landscape = ImportedLandscape(LandscapeFile = "land.txt", 
                          Resolution = 1000, 
                          HabPercent = TRUE, 
                          K_or_DensDep = 100, 
                          SpDistFile = "introduction.txt", 
                          SpDistResolution = 10000)

demo <- Demography(Rmax = 1.4)

distance = matrix(c(20000, 1000, 0.5 ), ncol = 3, byrow = T)

dispersal1 <-  Dispersal(Emigration = Emigration(EmigProb = 0.1), Transfer =  DispersalKernel(DoubleKernel = T, Distances = distance)) + Settlement()

init <- Initialise(InitType = 1, # = initialisation from a loaded species distribution map
                   SpType = 0,   # = all suitable cells within all distribution presence cells
                   InitDens = 0) # = at carrying capacity

sim <- Simulation(Simulation = 1,
                  Years = 1000,
                  Replicates = 2,
                  OutIntRange = 5,
                  OutIntPop = 5,
                  OutIntOcc = 5)

simulation1 <- RSsim() + landscape + demo + dispersal1 + sim + init
validateRSparams(simulation1) ## check parameter validity
```

[1] TRUE

My problem is that, when I start the simulation, I get the error that the

 *** Extents of Species Distribution map do not match headers of LandscapeFile
Error reading landscape ASCII haeders - aborting

This is despite specifying the same extent on both the landscape and Species distribution maps i.e

```{r, echo=TRUE, warning=FALSE, message=FALSE}
AllData@bbox = matrix(c(73.25117, -54.74922, 167.9904, -10.05176), nrow = 2)
introduction@bbox = matrix(c(73.25117, -54.74922, 167.9904, -10.05176), nrow = 2)
```

Additional information:

The original files used to create the maps had the columns:

a) Species distribution data

  | species | lon | lat | uncertainty | date | year | month
-- | -- | -- | -- | -- | -- | -- | --

b) Landscape suitability data

  | lon | lat | logistic_value 
-- | -- | -- | -- | -- 

How can I overcome this error and run the simulation successfully?

Regards,
Jean

**Comment by annekathleenmalchow** [comment, marked as answer]
Hi Jean,
thanks for your question. The error occurs because RangeShifter expects an [ESRI grid](https://en.wikipedia.org/wiki/Esri_grid) format. Note that this format assumes that the data is on a regular raster grid. In R, you can use the raster package to convert your data.frame with lat-lon data to a raster with the correct projection. For this, you can use the `rasterFromXYZ()` functions (but I also recommend you to read [this thread](https://gis.stackexchange.com/questions/20018/how-can-i-convert-data-in-the-form-of-lat-lon-value-into-a-raster-file-using-r).
Once you have a correct raster, you can export it to an Esri Grid file with
`writeRaster(r, format="ascii", filename = "Inputs/map_01"), datatype = 'INT1S', NAflag = -9, overwrite = F)`
I hope this helps!
All the best,
Anne

**Reply by Jeankwizera** [reply]
Hi Anne, thank you for the feedback. I was able to solve my problem with your guidance.
